src/nvim/mbyte.c

```diff
@@ -1064,17 +1020,25 @@ int utfc_ptr2len_len(const char_u *p, int size)
  */
 int utf_char2len(int c)
 {
-  if (c < 0x80)
-    return 1;
-  if (c < 0x800)
-    return 2;
-  if (c < 0x10000)
-    return 3;
-  if (c < 0x200000)
-    return 4;
-  if (c < 0x4000000)
-    return 5;
-  return 6;
+  int buf[4] = {0, 0, 0, 0};
+  switch (utf8proc_category(c)) {
+    case UTF8PROC_CATEGORY_LU:
```

Comment:
This is currently just a hack. And there will probably be a couple of it during this WIP. So don't be too confused when you see strange things. Input is certainly appreciated but keep in mind that this is basically me experimenting and getting to know the code base.